static/panes/diff.js

```diff
@@ -33,22 +33,50 @@ var ga = require('../analytics');
 require('../modes/asm-mode');
 require('selectize');
 
-function State(id, model) {
+var
+    DiffType_ASM = 0,
+    DiffType_CompilerStdOut = 1,
+    DiffType_CompilerStdErr = 2,
+    DiffType_ExecStdOut = 3,
+    DiffType_ExecStdErr = 4;
+
+function State(id, model, difftype) {
     this.id = id;
     this.model = model;
     this.compiler = null;
     this.result = null;
+    this.difftype = difftype;
 }
 
 State.prototype.update = function (id, compiler, result) {
     if (this.id !== id) return false;
     this.compiler = compiler;
     this.result = result;
-    var asm = result.asm || [];
-    this.model.setValue(_.pluck(asm, 'text').join("\n"));
+    this.refresh();
+
     return true;
 };
 
+State.prototype.refresh = function () {
+    var output = [];
+    if (this.result) {
+        if (this.difftype === DiffType_ASM) {
```

Comment:
Just a style thing: I'd probably use a `switch` here (I thikn JS lets us switch with non-const `case`s). Another possibility is to have a map of things like '{DiffType_CompilerStdErr: 'execResult'}` and then use `this.result[map[type]]`. This is absolutely perfect as it is: just giving some pre-coffee thoughts :grinning: